dojo_plugin/models/__init__.py

```diff
@@ -405,6 +405,14 @@ def resources(self, value):
             resource.resource_index = resource_index
         self._resources = value
 
+    @hybrid_method
+    def visible(self):
+        return len(self.visible_challenges()) > 0
+
+    @visible.expression
+    def visible(cls, when=None):
+        return len(cls.visible_challenges()) > 0
```

Comment:
Are you sure that this is valid sqlalchemy? At a glance it doesn't look like it would be since it iterates.

Reply:
I'm not, tbh I'm not familiar with sqlachemy. I just copied the `visible()` method of `DojoChallenges`